### PR TITLE
Pin Rustup's version

### PR DESCRIPTION
**Pull request by Mcat12**
Version updates of rustup can have negative effects on the image, not to mention the reproducibility of the docker container. For one such example of a negative effect, see #62.

### CODE REVIEW

**Comment by clux** [comment]
Thank you. That makes sense from a reproducibility standpoint, but it's going to be another thing _I_ manually have to bump. Is it possible to find the current latest version somewhere? That way we can stash the mechanism for updating the version inside the `update_libs.py`.

**Comment by Mcat12** [comment]
That's a good idea. From looking at the [self-update code], you can find the latest version at https://static.rust-lang.org/rustup/release-stable.toml. I'll look into adding that to the `update_libs.py` script.

[self-update code]: https://github.com/rust-lang/rustup.rs/blob/1a46072ac0a82d3a2dcd38a0afdcc2c7caa69507/src/cli/self_update.rs#L1501

**Comment by Mcat12** [comment]
@clux Do you want to use the version of `rustup` in the AUR or the upstream version?

**Comment by Mcat12** [comment]
I've added the Rustup update functionality into `update_libs.py`, though it adds a dependency on [`toml`](https://pypi.org/project/toml/) for parsing Rustup's version config. Is this ok?

File: update_libs.py [hunk 1]
```diff
@@ -65,13 +66,28 @@ def pkgver(package):
...
+    :return: The current Rustup version
+    """
+
+    req = urllib.urlopen('https://static.rust-lang.org/rustup/release-stable.toml')
+    metadata = toml.loads(req.read().decode("utf-8"))
+    req.close()
+
+    return metadata['version']
+
+
```
Comment (clux) [inline update_libs.py]: beautiful. thank you very much.

File: Dockerfile [hunk 2]
```diff
@@ -35,8 +35,12 @@ RUN apt-get update && apt-get install -y \
...
 # Install rust using rustup
 ARG CHANNEL="nightly"
-RUN curl https://sh.rustup.rs -sSf | \
-    sh -s -- -y --default-toolchain ${CHANNEL} --profile minimal && \
+ENV RUSTUP_VER="1.20.2" \
+    RUST_ARCH="x86_64-unknown-linux-gnu"
+RUN curl "https://static.rust-lang.org/rustup/archive/${RUSTUP_VER}/${RUST_ARCH}/rustup-init" -o rustup-init && \
+    chmod +x rustup-init && \
+    ./rustup-init -y --default-toolchain ${CHANNEL} --profile minimal && \
+    rm rustup-init && \
```
Comment (clux) [inline Dockerfile]: any particular reason why you saving this temporary script and not just piping it to `sh` ?

**Comment by clux** [comment]
Yeah, absolutely. Happy to merge this. Just one question on the Dockerfile.

File: Dockerfile (hunk 2 above)
Comment (Mcat12) [reply Dockerfile]: It's an executable, which I don't think you can pipe.

**Comment by Mcat12** [comment]
Re-triggering Travis as it may have run into a temporary error.

File: Dockerfile (hunk 2 above)
Comment (clux) [reply Dockerfile]: oh, wow it's a big binary. okay, makes sense then.

**Comment by clux** [comment]
Yeah, not sure what's up with master atm. Your README change clearly didn't break things :thinking:

**Comment by Mcat12** [comment]
I just ran though the Travis script locally on master and didn't have any issues. Maybe [debugging] it would help shed some light?

This could also be related to a recent change on Travis: https://changelog.travis-ci.com/xenial-and-cache-support-added-for-arm-builds-123563

[debugging]: https://docs.travis-ci.com/user/running-build-in-debug-mode/

**Comment by Mcat12** [comment]
@clux Please let me know if I can do anything further to help find the current CI issue.

**Comment by clux** [comment]
So, yesterday, I had a look to see if I could trigger debug builds, and it took me into a different travis interface that required payment, but then ultimately got distracted. Good news seems to be that the ephemeral travis issue fixed itself. :man_shrugging:

**Pull request merged**